apis/sprints_api.py

```diff
@@ -168,3 +176,23 @@ def check_duplicate_prompt():
     res = prompts.check_for_sprint_duplicates(start, end)
     return jsonify(res)
 
+# get the next hint
+@sprint_api.get('/hint')
+# @check_request_json({"start": str, "end": str})
+def get_hint():
+    start = request.args.get('start')
+    end = request.args.get('end')
+
+    print(start)
+    print(end)
+
+    if (start is None or end is None): return "Invalid Request", 400
+
+    # which algorithm to use?
+    # greedy = GreedySearch(embeddings_provider, graph_provider)
+    # path = greedy.search(start, end)
+
+    beam = BeamSearch(embeddings_provider, graph_provider)
```

Comment:
So here, what you should do instead of doing the whole search is do a single step, i.e. look at the pages linked from your current article, and then compare them with the embedding of the goal.

Doing a whole beam search is a lot of unnecessary work.

Note this would require modifying the wsbot library, which I can help with if you're intested. The best way I see of doing that is adding a separate `get_link_rankings` or something to the wsbot lib, that both the `Search` and external users (such as this API) can call.

Reply:
Ah that makes sense I'll look into this. Thanks for reviewing!